psydac/api/tests/test_2d_navier_stokes.py

```diff
@@ -98,17 +114,18 @@ def run_time_dependent_navier_stokes_2d(filename, dt_h, nt, newton_tol=1e-4, max
     # Reynolds number
     Re = 1e4
 
-    # define the linearized navier stokes
-    a = BilinearForm(((du,dp),(v, q)), integral(domain, dot(du,v) + dt*dot(Transpose(grad(du))*u, v) + dt*dot(Transpose(grad(u))*du, v)
-                                                      + dt*Re**-1*inner(grad(du), grad(v)) - dt*div(du)*q - dt*dp*div(v) + dt*1e-10*dp*q) )
-    l = LinearForm((v, q),             integral(domain, dot(u,v)-dot(u0,v) + dt*dot(Transpose(grad(u))*u, v)
-                                                      + dt*Re**-1*inner(grad(u), grad(v)) - dt*div(u)*q - dt*p*div(v) + dt*1e-10*p*q) )
+    F  = 0.5*dt*dot(Transpose(grad(u ))*u , v) + 0.5*dt*Re**-1*inner(grad(u ), grad(v)) - 0.5*dt*div(u )*q - 0.5*dt*p *div(v) + 0.5*dt*1e-10*p *q
+    F0 = 0.5*dt*dot(Transpose(grad(u0))*u0, v) + 0.5*dt*Re**-1*inner(grad(u0), grad(v)) - 0.5*dt*div(u0)*q - 0.5*dt*p0*div(v) + 0.5*dt*1e-10*p0*q
+    
+    l = LinearForm((v, q), integral(domain, dot(u,v)-dot(u0,v) + F + F0) )
+    a = linearize(l, (u,p), trials=(du, dp))
```

Comment:
It is hard to recognize the Navier-Stokes equations in these formulas. I think it is much easier to read if you define `F` as a lambda function of `(u, p)`, without the factor `0.5*dt`. Then the linear form will read
```python
l = LinearForm((v, q), integral(domain, dot(u - u0, v) + F(u, p) + F(u0, p0)))
```

Reply:
done